### PR TITLE
Recover updates to docs and release notes

### PR DESCRIPTION
**Pull request by alculquicondor**
#### What type of PR is this?

/kind documentation

#### What this PR does / why we need it:

I accidentally deleted some commits from #351 instead of squashing.

#### Which issue(s) this PR fixes:

Fixes #

### CODE REVIEW

File: docs/concepts/cluster_queue.md [hunk 1]
```diff
@@ -39,29 +39,29 @@ You can specify the quota as a [quantity](https://kubernetes.io/docs/reference/k
...
 In a ClusterQueue, you can define quotas for multiple [compute resources](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#resource-types)
-(cpu, memory, GPUs, etc.).
+(CPU, memory, GPUs, etc.).
 
-For each resource, you can define quotas for multiple _flavors_. A
-flavor represents different variations of a resource. The variations can be
-defined in a [ResourceFlavor object](#resourceflavor-object).
+For each resource, you can define quotas for multiple _flavors_.
+Flavors represent different variations of a resource (for example, different GPU
+models). A flavor can be defined using a [ResourceFlavor object](#resourceflavor-object).
```
Comment (ahg-g) [inline docs/concepts/cluster_queue.md:46]:
```suggestion
models). A flavor is defined using a [ResourceFlavor object](#resourceflavor-object).
```

File: docs/concepts/cluster_queue.md [hunk 2]
```diff
@@ -39,29 +39,29 @@ You can specify the quota as a [quantity](https://kubernetes.io/docs/reference/k
...
+This is typical for `cpu` and `memory`, where the flavors are generally tied to
+a machine family or VM availability policies. When two or more resources in a
+ClusterQueue match their flavors, they are said to be codependent resources.
 
-If this is the case, the resources in the ClusterQueue must list the same
-flavors in the same order. When two or more resources match their flavors,
-they are said to be codependent. During admission, for each pod set in a
-Workload, Kueue assigns the same flavor to the codependent resources that the
-pod set requests.
+To manage codependent resources, list the flavors in the ClusterQueue resources
```
Comment (ahg-g) [inline docs/concepts/cluster_queue.md:62]:
```suggestion
To manage codependent resources, you should list the flavors in the ClusterQueue resources
```

File: docs/concepts/cluster_queue.md [hunk 3]
```diff
@@ -238,16 +238,28 @@ ClusterQueue.
...
   ClusterQueue's `.spec.resources[*].flavors`. For each flavor, Kueue attempts
-  to fit a Workload's pod set using the `min` quota of the ClusterQueue or the
-  unused `min` quota of other ClusterQueues in the cohort, up to the `max` quota
-  of the ClusterQueue. If the workload doesn't fit, Kueue proceeds evaluating the next
-  flavor in the list.
-- A ClusterQueue can only borrow quota of flavors it defines and it can only
-  borrow quota for one flavor.
+  to fit a Workload's pod set according to the quota defined in the
+  ClusterQueue for the flavor and the unused quota in the cohort.
+  If the workload doesn't fit, Kueue proceeds evaluating the next flavor in the
```
Comment (ahg-g) [inline docs/concepts/cluster_queue.md:248]:
```suggestion
  If the workload doesn't fit, Kueue evaluates the next flavor in the
```

File: docs/concepts/cluster_queue.md (hunk 1 above)
Comment (alculquicondor) [reply docs/concepts/cluster_queue.md]: Done

File: docs/concepts/cluster_queue.md (hunk 2 above)
Comment (alculquicondor) [reply docs/concepts/cluster_queue.md]: Done

File: docs/concepts/cluster_queue.md (hunk 3 above)
Comment (alculquicondor) [reply docs/concepts/cluster_queue.md]: Done

**Pull request merged**